syntax = "proto3";

import "antiflicker_state.proto";
import "camera_synchronization_state.proto";
import "digital_crop_state.proto";
import "digital_event_mask_state.proto";
import "event_rate_control_state.proto";
import "event_rate_noise_filter_state.proto";
import "event_trail_filter_state.proto";
import "hw_register_state.proto";
import "ll_biases_state.proto";
import "region_of_interest_state.proto";
import "trigger_in_state.proto";
import "trigger_out_state.proto";

package DeviceSerialization;

message DeviceState {
    AntiflickerState afk_state = 1;
    CameraSynchronizationState cam_sync_state = 2;
    DigitalCropState digital_crop_state = 3;
    DigitalEventMaskState digital_event_mask_state = 4;
    EventRateControlState event_rate_control_state = 5;
    EventRateNoiseFilterState event_rate_noise_filter_state = 6;
    EventTrailFilterState event_trail_filter_state = 7;
    HWRegisterState hw_register_state = 8;
    LLBiasesState ll_biases_state = 9;
    RegionOfInterestState roi_state = 10;
    TriggerInState trigger_in_state = 11;
    TriggerOutState trigger_out_state = 12;
}
